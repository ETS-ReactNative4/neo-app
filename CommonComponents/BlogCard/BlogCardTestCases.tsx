import React from "react";

import { ITestCase } from "../../TypeInterfaces/TestCases/ITestCases";
import BlogCard from "./BlogCard";
import { ScrollView, Alert } from "react-native";

const data: ICardData[] = [
  {
    smallTitle: "blog",
    title: "10 amazing things to do in Europe",
    backgroundColor: "rgb(0, 109, 183)"
  },
  {
    smallTitle: "blog",
    title: "5 essential things to know about Dubai",
    backgroundColor: "rgb(206, 133, 48)"
  },
  {
    smallTitle: "guide",
    title: "A localâ€™s guide to experience Turkey",
    backgroundColor: "rgb(188, 74, 74)"
  }
];

interface ICardData {
  smallTitle: string;
  title: string;
  backgroundColor: string;
}

interface BlogCardWrapperProps {
  cardData: ICardData[];
}

const BlogCardWrapper = ({ cardData }: BlogCardWrapperProps) => {
  return (
    <ScrollView horizontal showsHorizontalScrollIndicator={false}>
      {cardData.map((dataObj, index) => {
        return (
          <BlogCard
            key={index}
            smallTitle={dataObj.smallTitle}
            title={dataObj.title}
            bgColor={dataObj.backgroundColor}
            action={() => Alert.alert(`Click -> ${dataObj.title}`)}
          />
        );
      })}
    </ScrollView>
  );
};

const BlogCardTestCases: ITestCase[] = [
  {
    title: "Block Card",
    Component: (
      <BlogCard
        smallTitle={"blog"}
        title={"10 amazing things to do in Europe"}
        action={() => Alert.alert("Click -> 10 amazing things to do in Europe")}
      />
    )
  },
  {
    title: "Multiple Block Cards",
    Component: <BlogCardWrapper cardData={data} />
  }
];

export default BlogCardTestCases;
