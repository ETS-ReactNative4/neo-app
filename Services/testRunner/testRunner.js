import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
