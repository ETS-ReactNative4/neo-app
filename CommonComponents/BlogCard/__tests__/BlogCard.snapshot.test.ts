import BlogCardTestCases from "../BlogCardTestCases";
import testRunner from "../../../Services/testRunner/testRunner";

BlogCardTestCases.map(testRunner);
