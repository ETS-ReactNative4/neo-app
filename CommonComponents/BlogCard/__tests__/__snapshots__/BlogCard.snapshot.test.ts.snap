// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block Card 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  onPress={[Function]}
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "flexDirection": "column",
          "height": 172,
          "justifyContent": "space-between",
          "marginRight": 12,
          "padding": 16,
          "width": 146,
        },
        Object {
          "backgroundColor": "rgba(83, 87, 109, 1)",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(247, 247, 247)",
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 11,
          "lineHeight": 11,
          "textTransform": "uppercase",
        }
      }
    >
      blog
    </Text>
    <Text
      ellipsizeMode="tail"
      numberOfLines={3}
      style={
        Object {
          "color": "rgb(247, 247, 247)",
          "fontFamily": "SourceSansPro-SemiBold",
          "fontSize": 16,
          "lineHeight": 20,
        }
      }
    >
      10 amazing things to do in Europe
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`Multiple Block Cards 1`] = `
<ScrollView
  horizontal={true}
  showsHorizontalScrollIndicator={false}
>
  <TouchableOpacity
    activeOpacity={0.8}
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "flexDirection": "column",
            "height": 172,
            "justifyContent": "space-between",
            "marginRight": 12,
            "padding": 16,
            "width": 146,
          },
          Object {
            "backgroundColor": "rgb(0, 109, 183)",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(247, 247, 247)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 11,
            "lineHeight": 11,
            "textTransform": "uppercase",
          }
        }
      >
        blog
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Object {
            "color": "rgb(247, 247, 247)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 16,
            "lineHeight": 20,
          }
        }
      >
        10 amazing things to do in Europe
      </Text>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.8}
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "flexDirection": "column",
            "height": 172,
            "justifyContent": "space-between",
            "marginRight": 12,
            "padding": 16,
            "width": 146,
          },
          Object {
            "backgroundColor": "rgb(206, 133, 48)",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(247, 247, 247)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 11,
            "lineHeight": 11,
            "textTransform": "uppercase",
          }
        }
      >
        blog
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Object {
            "color": "rgb(247, 247, 247)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 16,
            "lineHeight": 20,
          }
        }
      >
        5 essential things to know about Dubai
      </Text>
    </View>
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.8}
    onPress={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "flexDirection": "column",
            "height": 172,
            "justifyContent": "space-between",
            "marginRight": 12,
            "padding": 16,
            "width": 146,
          },
          Object {
            "backgroundColor": "rgb(188, 74, 74)",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(247, 247, 247)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 11,
            "lineHeight": 11,
            "textTransform": "uppercase",
          }
        }
      >
        guide
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={3}
        style={
          Object {
            "color": "rgb(247, 247, 247)",
            "fontFamily": "SourceSansPro-SemiBold",
            "fontSize": 16,
            "lineHeight": 20,
          }
        }
      >
        A localâ€™s guide to experience Turkey
      </Text>
    </View>
  </TouchableOpacity>
</ScrollView>
`;
